### ISSUE TITLE
print(ggplot_obj) not returning a ggplot object

### ISSUE THREAD

**Issue by gonike** [hadley/ggplot2#1087]
The documentation of `print.default` states that 

     'print' prints its argument and returns it _invisibly_ (via
     'invisible(x)').

Hence I expected the following to work:

    plot_obj <- ggplot(data = data.frame(x = 1:10, y = 1:10), aes(x = x, y = y)) + geom_point()
    plot_obj2 <- print(plot_obj)
    print(plot_obj2 + ggtitle("testplot"))

... but it doesn't (`Error in plot_obj2 + ggtitle("testplot") : non-numeric argument to binary operator`), since `plot_obj2` is of class `list` and not `ggplot` (as is `plot_obj`).

A quick look at [plot-render.r](https://github.com/hadley/ggplot2/blob/4bb9270ef4d5d5062353438fd99d17b6f6de98a2/R/plot-render.r#L179-L195) confirms that `print.ggplot(x)` does not return `x` but rather `ggplot_build(x)` (invisibly).

So to summarise,
- the behavior of `print.ggplot` is not in tune with the documentation of `print.default` and
- it is not documented.

A little bit of context on why I wanted to do this: I have various custom functions that construct ggplot objects and print them. I found it very convenient to just have `print(obj)` as the final statement in my function, believing that this would also invisibly return the object, s.t. I could further modify it outside of the function if necessary. Now I have to do `print(obj); invisible(obj)`.

Is there a reason for this choice of implementation? If yes, I'd think it should at least be documented.

**Comment by hadley** [comment]
@wch does the existing behaviour need to be preserved for shiny?
